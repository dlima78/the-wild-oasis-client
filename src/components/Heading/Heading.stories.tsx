import type { Meta, StoryObj } from '@storybook/react'

import Heading, { HeadingProps } from '.'

const meta = {
  title: 'Heading',
  component: Heading,
  argTypes: {
    type: {
      options: ['h1', 'h2', 'h3'],
      control: { type: 'select' }
    }
  }
} as Meta<HeadingProps>

export default meta
type Story = StoryObj<typeof Heading>

export const Default: Story = {
  args: {
    children: 'The Wild Oasis'
  }
}
