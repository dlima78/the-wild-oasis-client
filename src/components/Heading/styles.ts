import { css, styled } from 'styled-components'
import { DefaultTheme } from 'styled-components/dist/types'
import { HeadingProps } from '.'

export type WrapperProps = Pick<HeadingProps, 'type'>

const WrapperModifiers = {
  h1: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xxlarge};
    font-weight: 600;
  `,
  h2: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xlarge};
    font-weight: 600;
  `,
  h3: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xlarge};
    font-weight: 500;
  `
}

export const Wrapper = styled.h1<WrapperProps>`
  ${({ theme, type }) => css`
    line-height: 1.4rem;

    ${!!type && WrapperModifiers[type](theme)}
  `}
`
