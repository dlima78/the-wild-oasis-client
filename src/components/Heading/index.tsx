import * as S from './styles'

export type HeadingProps = {
  children: React.ReactNode
  type?: 'h1' | 'h2' | 'h3'
}

const Heading: React.FC<HeadingProps> = ({ type = 'h1', children }) => (
  <S.Wrapper as={type}>{children}</S.Wrapper>
)

export default Heading
