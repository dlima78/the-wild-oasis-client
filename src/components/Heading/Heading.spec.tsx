import { screen } from '@testing-library/react'

import { renderWithTheme } from '@/utils/tests/helpers'
import Heading from '.'

describe('<Heading>', () => {
  it('should render h1 heading by default', () => {
    const { container } = renderWithTheme(<Heading>The Wild Oasis</Heading>)

    expect(screen.getByRole('heading', { level: 1 })).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render h2 heading by default', () => {
    renderWithTheme(<Heading type="h2">The Wild Oasis</Heading>)

    expect(screen.getByRole('heading', { level: 2 })).toBeInTheDocument()
  })

  it('should render h3 heading by default', () => {
    renderWithTheme(<Heading type="h3">The Wild Oasis</Heading>)

    expect(screen.getByRole('heading', { level: 3 })).toBeInTheDocument()
  })
})
