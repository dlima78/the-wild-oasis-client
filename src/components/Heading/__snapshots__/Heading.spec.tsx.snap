// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading> should render h1 heading by default 1`] = `
.c0 {
  line-height: 1.4rem;
}

<h1
  class="c0"
>
  The Wild Oasis
</h1>
`;
